contracts/eden/src/atomicassets.cpp

```diff
@@ -41,9 +41,15 @@ namespace eden::atomicassets
                                      int32_t template_id)
    {
       ::atomicassets::templates_t templates(contract, collection.value);
-      const auto& templ = templates.get(template_id);
+      const auto& templ = templates.get(
+          template_id,
+          std::string("fail to read template id " + std::to_string(template_id)).c_str());
```

Comment:
You don't need a cast.  It's already a string.

Reply:
I cleaned them across the board